packages/embark/src/lib/modules/whisper/index.js

```diff
@@ -203,22 +202,19 @@ class Whisper {
         'ws',
         '/embark-api/communication/listenTo/:topic',
         (ws, req) => {
-          self.webSocketsChannels[req.params.topic] = listenTo({
-            topic: req.params.topic,
-            messageEvents,
-            toHex: self.web3.utils.toHex,
+          const obs = listenTo({
             toAscii: self.web3.utils.hexToAscii,
+            toHex: self.web3.utils.toHex,
+            topic: req.params.topic,
             sig,
-            symKeyID,
-            subscribe: self.web3.shh.subscribe
-          }, (err, result) => {
-            if (ws.readyState === ws.CLOSED) {
-              return;
-            }
-            if (err) {
-              return ws.status(500).send(JSON.stringify({error: err}));
-            }
-            ws.send(JSON.stringify(result));
+            subscribe: self.web3.shh.subscribe,
+            symKeyID
+          }).pipe(takeUntil(fromEvent(ws, 'close').pipe(map(() => (
+            delete self.webSocketsChannels[req.params.topic]
+          )))));
```

Comment:
Nice! This is already better @michaelsbradleyjr 

One tiny suggestion:  I think `map()` should really be used to map from one signal to something else. What we do here however is creating a side effect (deleting that channel). Using the `tap()` operator instead of `map()` might be a better semantic fit here. 

Frankly it won't make a difference in the outcome because `takeUntil()` also just cares about whether it gets a signal or not, which it will get as `map()` emits probably 'null' or 'undefined' in the current case. 

Anyways, I thought it might be also a bit "less confusing" for other ppl that look at the code and try to understand why this has been `map`ped.